src/main/java/strategy/SupplyOperation.java

```
@@ -0,0 +1,13 @@
+package strategy;
+
+import core.basesyntax.FruitTransaction;
+import db.Storage;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
```

The task requires handling invalid input data, such as negative quantities. You should add a check here to ensure `transaction.getQuantity()` is not negative. A supply with a negative value is invalid and should throw an exception.